doc/tools/make_rst.py

```diff
@@ -731,6 +731,10 @@ def main() -> None:
 
         if os.path.basename(path) in ["modules", "platform"]:
             for subdir, dirs, _ in os.walk(path):
+                # Exclude godot-cpp test project files for the case of
+                # third-party modules with godot-cpp inside of them.
+                if "/godot-cpp/test" in subdir:
```

Comment:
Will this work even if the "godot-cpp" directory isn't at the top level?

I know most projects put it there, but I've got a couple that put it under "thirdparty" and other folks may do other things as well. Maybe we need a way to provide a list of exclude directories?

Reply:
Yes, in fact that's how I use it. This is just a string match, it will work regardless of how deep it is.

Reply:
Do paths returned by `os.walk()` on Windows use forward slashes? If not, this check won't work on Windows.

Reply:
I'm not fond of hardcoding something specific to a thirdparty repository here.
A project might use godot-cpp but with a different folder name (e.g. it may be cloned as `godot-cpp.git`), and the `test` folder may not always be there in the future or include documentation.

I think this would be better solved by improving the logic so it doesn't go so deep.

The patterns to support are:
```
modules/<name>/doc_classes
platform/<name>/doc_classes
```

So if we limit ourselves to one level of depth it should be fine I expect.

Otherwise, adding a way to pass excludes from the command line might be better than hardcoding exceptions here.

Reply:
@akien-mga In my module I am using `modules/4d/addons/4d/doc_classes`. Thoufh it is possible to infer the maximum depth level if we read the modules config.py files, that would be a larger change than I originally intended.

Reply:
@Calinou Good catch, it actually returns double-backslashes, so we need `.replace("\\\\", "/")` to fix it.

Reply:
@akien-mga Also I should mention that godot-cpp is not third-party, this is a Godot-maintained repository.